<launch>

  <param name="/use_sim_time" value="true" /> <!-- true: offline (rosbag); false: online -->
  <param name="robot_description" textfile="$(find cartographer_ros)/urdf/mid360_nanjing.urdf" />

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

  <node name="cartographer_node" pkg="cartographer_ros" type="cartographer_node" 
    args="-configuration_directory $(find cartographer_ros)/configuration_files
          -configuration_basename mid360_localization_nanjing.lua
          -load_state_filename /home/nvidia/wjd/testmaps/carto_map.pbstream"
    output="screen">

    <remap from="scan" to="front_scan" />
    <remap from="odom" to="odom" />
    <remap from="points2" to="points_calibrated"/>
    <remap from="imu" to="livox/imu_192_168_1_196" />

  </node>

  <node name="map_server" pkg="map_server" type="map_server" args="/home/nvidia/wjd/testmaps/carto_map.yaml" />


<!--0: 建图的初始位置作为导航的零点
    1：设置参数服务器参数作为导航的零点
    2：rviz设置初始位置作为导航的零点(本质是接收/initialpose这个topic，所以gps初始定位也用2)
-->
  <param name="/use_localization" type="int" value = "1"/>
  <param name="/set_inital_pose_x" type="double" value = "-36.02"/>
  <param name="/set_inital_pose_y" type="double" value = "-54.57"/>
  <param name="/set_inital_pose_z" type="double" value = "0.0"/>
  <param name="/set_inital_pose_ox" type="double" value = "0.0"/>
  <param name="/set_inital_pose_oy" type="double" value = "0.0"/>
  <param name="/set_inital_pose_oz" type="double" value = "0.362"/>
  <param name="/set_inital_pose_ow" type="double" value = "0.931"/>


  <node name="cartographer_occupancy_grid_node" pkg="cartographer_ros"
    type="cartographer_occupancy_grid_node" args="-resolution 0.05 -pure_localization 1"/>
    

  <node name="rviz" pkg="rviz" type="rviz" required="true"
      args="-d $(find cartographer_ros)/configuration_files/demo_2d.rviz" />


  <node name="playbag" pkg="rosbag" type="play" args="--clock $(arg bag_filename)" />


</launch>
